routes/webhooks.js

```diff
@@ -59,32 +62,38 @@ router.post('/pages.json', function(req, res, next) {
             })
             .done(function() {
                 // Move from workingDir to pages dir
-                var finalRepoPath = path.resolve(config.deploy.publicPagesDir, projectNamespace, projectName);
+                var siteRoot =
+                    path.resolve(config.deploy.publicPagesDir, projectNamespace, projectName);
+
                 // Delete workingDir
-                rmdir(finalRepoPath, function() {
-		    mkdocs_path = path.resolve(repoPath, 'mkdocs.yml');
-		    fs.exists(mkdocs_path, function (exists) {
-			if (exists) {
-			    mkdocs = YAML.load(mkdocs_path);
-			    mkdocs['site_dir'] = finalRepoPath;
-			    fs.writeFile(mkdocs_path, YAML.stringify(mkdocs, 4));
-			    var cmd = "(cd "+repoPath+" && mkdocs build --clean)";
-			    console.log('Building site with mkdocs') 
-			} else {
-			    // jekyll build --safe --source .tmp/Glavin001/gitlab-pages-example/ --destination pages/Glavin001/gitlab-pages-example
-			    var cmd = "jekyll build --safe --source \""+repoPath+"\" --destination \""+finalRepoPath+"\"";
-			}
-			exec(cmd, function (error, stdout, stderr) {
-                            // output is in stdout
-                            console.log('Done deploying '+projectNamespace+'/'+projectName);
-			});
-		    });
-                    // mv(repoPath, finalRepoPath, {
-                    //     mkdirp: true,
-                    //     clobber: true
-                    // }, function(err) {
-                    //     console.log('Done deploying '+projectNamespace+'/'+projectName);
-                    // });
+                rmdir(siteRoot, function() {
+                    mkdocs_path = path.resolve(repoPath, 'mkdocs.yml');
+                    npm_path = path.resolve(repoPath, 'package.json');
+                    var cmd =
+                      "jekyll build --safe --source \"" +
+                      repoPath + "\" --destination \"" +
+                      siteRoot + "\"";
+                    if (fs.existsSync(mkdocs_path)) {
+                        var mkdocs = YAML.load(mkdocs_path);
+                        mkdocs.site_dir = siteRoot;
+                        fs.writeFile(mkdocs_path, YAML.stringify(mkdocs, 4));
+                        cmd = "(cd "+repoPath+" && mkdocs build --clean)";
+                        console.log('Building site with mkdocs');
+                    } else if (fs.existsSync(npm_path)) {
+                        var npmConfig = require(npm_path);
```

Comment:
I wonder what happens when the server runs for a long time and we require in tons of stuff

Did you test what happens when you trigger the same repo twice (i.e. require the same `npm_path` twice)